Monika After Story/game/script-topics.rpy

```diff
@@ -3709,10 +3709,59 @@ label monika_penname:
     m 3b "Pen names also help writers create something totally different from their usual style of writing."
     m 3d "It really gives the writer the protection of anonymity and gives them a lot of creative freedom."
-    if mcname.lower() != player.lower():
-        m 2c "Is '[mcname]' a pseudonym that you're using?"
-        m "You're using two different names after all."
-        m 2d "'[mcname] and [player].'"
+    m "Do you have a pen name, [player]?"
+    menu:
+        "Yes":
+            m 1 su "Really? That's so cool!" 
+            m 1b "Do you care to tell me what it is?"
+            menu: 
+                "No, not at all.":
+                    $ penbool = False
+                    while not penbool:
+                        $ penname = renpy.input("What is your penname?",length=20).strip(' \t\n\r')
+                        $ lowerpen = penname.lower()
+                        if lowerpen == player.lower():
+                            m 3d "Oh, so you're using your pen name?" 
+                            m "I'd like to think we are on a first name basis with each other. We are dating, after all."
+                            m "But I guess it's pretty special that you shared your pen name with me!"
+                            $ penbool = True
+                        elif lowerpen =="sayori":
+                            m 2c "..."
+                            m 1n "...I mean, I won't question your choice of pennames, but..."
+                            m "If you want to name yourself after a character in this game, you should pick me!"
+                            $ penbool = True
+                        elif lowerpen =="natsuki":
+                            m 2c "..."
+                            m 1n "Well, I guess I shouldn't assume that you named yourself after {i}our{/i} Natsuki."
+                            m "It's something of a common name." 
+                            m "You might make me jealous, though."
+                            $ penbool = True
+                        elif lowerpen == "yuri":
+                            m 2c "..."
+                            m 1n "Well, I guess I shouldn't assume that you named yourself after {i}our{/i} Yuri."
+                            m "It's something of a common name."
+                            m 1i "Of course, there's something else that name could refer too..."
+                            if persistent.gender =="F":
+                                m 5 "And well...I could get behind that, since it's you~"
+                            $ penbool = True                                
+                        elif lowerpen =="monika":
+                            m "..."
+                            m 1 ebla "Aww, did you pick that for me?" 
+                            m "Even if you didn't, that's so sweet!"
+                            $ penbool = True   
+                        else:
```

Comment:
you may also want to check for an empty string and probably the specific case saying nevermind or something similar just in case player change their mind and decide not to say their pen name.
```
                elif lower pen == "nevermind": # in case player doesn't want to say their penance
                    # appropriate dialogue here
                    $ penbool = True
                elif not lowername: #check for empty string
                    # appropriate dialog here, also don't set penbool to true
```